src/Utilities/Compiler/Options/AbstractCategorizedAnalyzerConfigOptions.cs

```diff
@@ -88,13 +89,23 @@ public bool TryGetOptionValue<T>(string optionName, OptionKind kind, DiagnosticD
             }
         }
 
-        private (bool found, object? value) ComputeOptionValue<T>(string optionName, OptionKind kind, DiagnosticDescriptor rule, TryParseValue<T> tryParseValue)
+        private (bool found, object? value) ComputeOptionValue<T>(string optionName, OptionKind kind, DiagnosticDescriptor? rule, TryParseValue<T> tryParseValue)
         {
             var optionKeyPrefix = MapOptionKindToKeyPrefix(kind);
-            if (TryGetSpecificOptionValue(rule.Id, optionKeyPrefix, out var optionValue) ||
-                TryGetSpecificOptionValue(rule.Category, optionKeyPrefix, out optionValue) ||
-                TryGetAnySpecificOptionValue(rule.CustomTags, optionKeyPrefix, out optionValue) ||
-                TryGetGeneralOptionValue(optionKeyPrefix, out optionValue))
+
+            T optionValue;
```

Comment:
Out of curiosity, is it to reduce allocation that you declare the variable before rather than doing `out var`?

Reply:
No, the var is again needed in a separate if check below, hence it must be scoped to the whole method.